### DISCUSSION TITLE
using the script on an existing octoprint.

### DISCUSSION THREAD

**Discussion by meagodoy** [paukstelis/octoprint_deploy#170] [Q&A]
Hello,
Thanks for the great script. My question is I already have octoprint all configured running on a Linux server. I can use this octoprint as a template or I need to create a new template through the script before making copies.

**Comment by paukstelis** [comment]
> Hello, Thanks for the great script. My question is I already have octoprint all configured running on a Linux server. I can use this octoprint as a template or I need to create a new template through the script before making copies.

You would have to make a new instance to use as a template. octoprint_deploy needs to have documented that instance. However, you can easily copy everything from your old instance to the new template.

First, copy everything in your existing octoprint folder (presumably this is .octoprint) to something else:
`cp -r ~/.octoprint old_template`
Then remove all traces of the old installation:
`sudo octoprint_deploy/octoprint_deploy.sh remove`
Do system prepare and make a new instance with octoprint_deploy
`cp -r ~/old_template/* ~/.newinstancename`
